pkg/pillar/hypervisor/kvm.go

```diff
@@ -1181,23 +1181,35 @@ func getQmpListenerSocket(domainName string) string {
 func (ctx KvmContext) VirtualTPMSetup(domainName, agentName string, ps *pubsub.PubSub, warnTime, errTime time.Duration) error {
 	if ps != nil {
 		wk := utils.NewWatchdogKick(ps, agentName, warnTime, errTime)
-		return requestVtpmLaunch(domainName, wk, swtpmTimeout)
+		domainUUID, _, _, err := types.DomainnameToUUID(domainName)
+		if err != nil {
+			return fmt.Errorf("failed to extract UUID from domain name: %v", err)
+		}
+		return requestVtpmLaunch(domainUUID.String(), wk, swtpmTimeout)
```

Comment:
Nit, but if the argument type was uuid.UUID and not a string then the compiler would help check.